File: Villainous.ClientCmd/ClientRegistration.cs
namespace Villainous.ClientCmd;

public class ClientRegistration
{

    public static async Task<LoginInfo> Register(ILogger logger, VillainousClientConfig config)
    {
        var username = config.Username;
        if (username == null)
        {
            await Console.Out.WriteLineAsync("What's your username?");
            username = await Console.In.ReadLineAsync();
        }

        logger.Print($"Welcome {username}!");

        var httpClient = new HttpClient { BaseAddress = new Uri(config. BaseAddress) };

        var loginInfo = await httpClient.PostJson("User/Register", new { Username = username, Password = "SecurePasswordsAreImportant!!!111OneEleven" }).GetAsJson<LoginInfo>();
        httpClient.AuthenticateWithBearer(loginInfo.AccessToken);
        logger.LogDebug("Registered!");
        return loginInfo;
    }
}

public record LoginInfo(Guid Id, string AccessToken);

File: villainous.client.react/src/ClientRegistration.ts
import { Guid } from './Contracts';
import { VillainousClientConfig } from './GameContext';


export class ClientRegistration {
    static Register = async (config: VillainousClientConfig) => {
        const response = await fetch(`${config.baseAddress}/User/Register`, {
            method: 'POST',
            headers: {
                "Content-Type": "application/json",
            },
            body: JSON.stringify({
                Username: "Willy",
                Password: "SecurePasswordsAreImportant!!!111OneEleven"
            })
        });

        return await response.json() as LoginInfo;
    };
}

export type LoginInfo = {
    id: Guid;
    accessToken: string;
}
